docs/firefighting/index.md

```diff
@@ -0,0 +1,787 @@
+
+##What do do when one or more Riak nodes has a corrupt database
+AAE error seen (logged repeatedly) in /var/log/riak/console.log accompanied by very high CPU load:
+
+in /var/log/riak/console.log:
+```
+2017-04-25 13:18:01.615 [info] <0.824.0>@riak_kv_vnode:maybe_create_hashtrees:234 riak_kv/890602560248518965780370444936484965102833893376: unable to start index_hashtree: {error,{{badmatch,{error,{db_open,"Corruption: truncated record at end of file"}}},[{hashtree,new_segment_store,2,[{file,"src/hashtree.erl"},{line,725}]},{hashtree,new,2,[{file,"src/hashtree.erl"},{line,246}]},{riak_kv_index_hashtree,do_new_tree,3,[{file,"src/riak_kv_index_hashtree.erl"},{line,712}]},{lists,foldl,3,[{file,"lists.erl"},{line,1248}]},{riak_kv_index_hashtree,init_trees,3,[{file,"src/riak_kv_index_hashtree.erl"},{line,565}]},{riak_kv_index_hashtree,init,1,[{file,"src/riak_kv_index_hashtree.erl"},{line,308}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,304}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,239}]}]}}
+2017-04-25 13:18:01.633 [error] <0.1988.0> CRASH REPORT Process <0.1988.0> with 0 neighbours exited with reason: no match of right hand value {error,{db_open,"Corruption: truncated record at end of file"}} in hashtree:new_segment_store/2 line 725 in gen_server:init_it/6 line 328
+```
+
+The resolution is to delete the AAE hash tree as recommended by http://stackoverflow.com/a/28479640/10840:
+
+```
+ansible@hqriak18:~$ sudo service riak stop # probably not necessary, just restart after moving, maybe restart not even needed?
+ansible@hqriak18:~$ sudo mkdir -p sudo mkdir /opt/data/ecrypt/zjunk/20170425
+ansible@hqriak18:~$ sudo mv /opt/data/ecrypt/riak/anti_entropy/890602560248518965780370444936484965102833893376 /opt/data/ecrypt/zjunk/20170425/
+ansible@hqriak18:~$ mkdir /opt/data/ecrypt/zjunk/20170425/v0
+ansible@hqriak18:~$ sudo mv /opt/data/ecrypt/riak/anti_entropy/v0/890602560248518965780370444936484965102833893376 /opt/data/ecrypt/zjunk/20170425/v0/
+ansible@hqriak18:~$ sudo service riak start
+```
+
+Sometimes node data can become corrupt in a way that prevents the riak service from starting. In this case the node will begin to restart, but will crash while reading indexes from the disk into RAM. One example error we have seen in the past:
+
+In /var/log/riak/console.log
+```
+2016-09-28  00:02:22.350 [error] <0.638.0>@riak_kv_vnode:init:497 Failed to  start riak_cs_kv_multi_backend backend for index  1255977969581244695331291653115555720016817029120 error:  [{riak_kv_eleveldb_backend,{db_open,"Corruption: truncated record at end  of file"}}]
+``
+`
```

Comment:
Bad ending to code block. I didn't review very carefully beyond this point because the formatting is pretty messed up in the [markdown view](https://github.com/dimagi/commcare-cloud/blob/8fcd91a2958e90d2b3a6ccb514cb0d86fdab5bf9/docs/firefighting/index.md).

Reply:
Thanks for finding. I think i tracked down some other issues as well

https://github.com/dimagi/commcare-cloud/pull/2612/commits/db92b6a927fac10ced3592a0a3447f34f3ad7861
https://github.com/dimagi/commcare-cloud/pull/2612/commits/30863739a14da510a4a423d5407e42461509ed99
https://github.com/dimagi/commcare-cloud/pull/2612/commits/3f1ede6db1185481c527597ba3a51801f7f491fd